airflow/bin/cli.py

```diff
@@ -361,13 +362,87 @@ def run(args, dag=None):
 
     task = dag.get_task(task_id=args.task_id)
     ti = TaskInstance(task, args.execution_date)
-    ti.refresh_from_db()
-    if ti.current_state() == "failed":
-        # This behavior is always correct (failed tasks should not run), but is implemented to specifically handle the
-        # case where a worker goes down, the celery task gets re-queued, and the scheduler kills the airflow task
-        # without knowing to revoke the task from the celery queue.
+    ti.refresh_from_db(include_job_id=True)
```

Comment:
This is the beginning of the "new worker" side of the scheduler bounce recovery scenario.  Most of the new logic is here.  This is in `cli.py - run()` which is what airflow workers do when they are given a task instance to execute.